drivers/spi/spi_ll_stm32.c

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SPI_LEVEL
+#include <logging/sys_log.h>
+
+#include <misc/util.h>
+#include <kernel.h>
+#include <board.h>
+#include <errno.h>
+#include <spi.h>
+
+#include <clock_control/stm32_clock_control.h>
+#include <clock_control.h>
+
+#include <drivers/spi/spi_ll_stm32.h>
+#include <spi_ll_stm32.h>
+
+#define CONFIG_CFG(cfg)						\
+((const struct spi_stm32_config * const)(cfg)->dev->config->config_info)
+
+#define CONFIG_DATA(cfg)					\
+((struct spi_stm32_data * const)(cfg)->dev->driver_data)
+
+#ifdef CONFIG_SPI_STM32_INTERRUPT
```

Comment:
Enforce Async as mandatory seems dangerous since I know some SPI HW not using IRQs at all, yes it's not the case for STM32 for sure, but I really think this should be mandatory.

Is it blocker to leave non-interrupt support ?